### ISSUE TITLE
Gtk Port: Character More info bug

### ISSUE THREAD

**Issue by TheShadowOfHassen** [olivierkes/manuskript#1107]
So for my first addition to the Gtk port I decided to implement a feature that I've wanted for years. A template for Your More Details in character. It's the listbox thing. I always found that it really saps your writing momentum when you originally start with the characters and then take ten minutes to set someones eye color/ hair color etc.

So to my point. I noticed that in the gtk port manuskript uses a dictionary to hold its more info `{DataName:DataValue}` the original had two inside lists `[[DataName,DataValue]]` This is less a bug and more a potential bug when we get saving done, unless you count my template and then its a bad bug. Dictionaries allow only one name for each thing so no `{DataName:Eggs, DataName:Bacon}` That makes sense because you wouldn't have Hair color:Red Hair color: Blondm, but its not intuitive. That is way to much of a learning curve for anyone and frankly it'd be easier to fix than add a warning checker if you add two or more with the same name.

@TheJackiMonster If this is a problem and you don't have some brilliant design scheme I'll fix it within the next few days and put it in my pull request with the Template and the other bug fixes I've made. I just wanted to be sure though before I started.

**Comment by TheJackiMonster** [comment]
Using a dictionary for those values is an intentional design choice. The reason for this is that you can't refer to any attribute otherwise because they wouldn't have a unique identifier. So I was thinking about referencing character attributes in text or descriptions and it didn't make much sense to not implement it with a dictionary then.

Obviously we need to add warnings here avoiding that users create duplicates accidentally and data gets dropped during saving process. But the UI allows temporarily that multiple values under the same name are stored. So showing a warning in UI should be fine as well as a dialog to rename attributes or merging their values when saving.

**Comment by TheShadowOfHassen** [comment]
OK I think I know how to fix this. I'll add a check through on the code so that if you have a duplicate atrabute it'll just automatically put a 2 or 3 or etc at the end to clue people in. It'd fix any saving errors also. I'll add it to my pull request I'm working on.

**Comment by TheJackiMonster** [comment]
Hmmm, unsure if that's ideal. It's certainly a way many applications deal with that issue. However I don't think that's a solution for plotting anything with exact custom properties. I would mark all matching attributes in red or orange, show a notification with a warning and add a dialog when saving explaining it will cause an issue if continued.

I mean two attributes with the same name don't cause an actual error on saving. It's just wrong by design. If people really want to assign multiple values to one attribute, we should support that differently. For example we could support creating a list of values in the table. But I don't think this has priority.

However when renaming stuff automatically it will likely cause confusion. Because if referring to any attribute is just more difficult when you have multiple of them around with a number in the end. Users will always check back which is the proper one.

Do you have any usecase why users would create multiple attributes with the same name?

**Comment by TheShadowOfHassen** [comment]
The idea with the fix is that I am assuming most users would have only one name unless they got distracted and added a lot of new ones. But just in case it would fix any problems.

The data of a dictionary won't store two attributes which is what is the big problem. 

I think manuskript's indented user-base should be the beginners and red error signs and things about duplicate messages are a bit confusing. 

I already implemented the fix in my branch however. It can be a temporary fix until we get it sorted out.

Also doesn't manuskript auto save? I've written three books with manuskript and haven't pressed save once. A warning on save wouldn't work at all with autosave.

**Comment by TheJackiMonster** [comment]
> Also doesn't manuskript auto save? I've written three books with manuskript and haven't pressed save once. A warning on save wouldn't work at all with autosave.

Technically the Qt version did, yes. But the GTK version is a whole rewrite of the old data model as well to allow separating save & load process for each part of the project individually. So when the attributes of a character change, only that character needs to be saved. That's why a new auto-save functionality could probably only save changes in a fairly manner, for example when you switch panels or tabs during plotting. It doesn't need to save every time a user changes a character in an input widget as before (at least not for anything besides text areas).

It's also the goal to apply an improved process spell checking and saving only new parts when writing huge text.

**Comment by TheShadowOfHassen** [comment]
Yes, a revamped of the save version would be smart (and efficent but it doesn't make sense how an error message can still display during autosave) Maybe a warning pop up when you click away but I don't know if that's the best idea.
